platform/linux-generic/odp_ipsec.c

```diff
@@ -558,6 +559,7 @@ static int ipsec_in_iv(odp_packet_t pkt,
 		return -1;
 
 	if (ipsec_sa->aes_ctr_iv) {
+		ODP_STATIC_ASSERT(IPSEC_MAX_IV_LEN >= 16, "IPSEC_MAX_IV_LEN too small");
```

Comment:
These (also next patch) could be more useful near the constant definitions as it's just a plain constant comparison (unlike previous patch where a variable name was involved).

Reply:
I think it is better to have the asserts next to the code which the assertion is for. Otherwise the connection between the assert and the assumption in the code is easily lost and it would be easier to forget to update the assert when the code is changed. And if not forgotten, it would be more difficult to see whether other parts of code rely on the same assert. So, even though a static assert could be moved elsewhere, I think it shouldn't.

Reply:
For simple compile time asserts, I don't think there's really any connection loss as the connection is the usage of the constant itself. Additionally, the code gets a bit bloated as the same check is now in many places. In rest of linux-gen codebase, the static asserts seem to be immediately following or very close to the relevant type/constant definitions.

Reply:
> I don't think there's really any connection loss as the connection is the usage of the constant itself.

But you do not know which usages make the assumptions that are checked by the assert without reviewing all code that uses the identifiers that appear in the assert. To avoid the loss of context, one could add comments near the assert to explain which code depend on it. It is just simpler (and more maintainable, IMHO) to put the assert next to the code that depends on it. Similarly, if someone removes a piece of code, then the asserts that are no longer needed would go away with the removed code without extra pain. Globally scoped asserts could easily be missed, and even if not, updating or removing them would require careful code review of all other code that could potentially still depend on them.

I do not think that the fact that an assert can be changed to a compile time static assert means that it should be moved elsewhere from its context.

> In rest of linux-gen codebase, the static asserts seem to be immediately following or very close to the relevant type/constant definitions.

That is not necessarily always a good thing.

Reply:
Well, agree to disagree, resolving this.